mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -83,14 +85,19 @@ func (sp *StateProcessor[
 	st BeaconStateT,
 	dep DepositT,
 ) error {
-	depositIndex, err := st.GetEth1DepositIndex()
-	if err != nil {
-		return err
+	var nextDepositIndex uint64
+	switch depositIndex, err := st.GetEth1DepositIndex(); {
+	case err == nil:
+		nextDepositIndex = depositIndex + 1
+	case sp.processingGenesis && err != nil:
+		// Eth1DepositIndex may have not been set yet
```

Comment:
personal preference for readability:

```suggestion
	case err != nil && sp.processingGenesis:
		// If errored and still processing genesis, Eth1DepositIndex may have not been set yet
```

Reply:
Done